src/content/docs/en/guides/server-side-rendering.mdx

```diff
@@ -230,4 +230,64 @@ if (!product) {
 ### Server Endpoints
 
 A server endpoint, also known as an **API route**, is a special function exported from a `.js` or `.ts` file within the `src/pages/` folder. 
-The function takes an [endpoint context](/en/reference/api-reference/#endpoint-context) and returns a [Response](https://developer.mozilla.org/en-US/docs/Web/API/Response). A powerful feature of SSR, API routes are able to securely execute code on the server. To learn more, see our [Endpoints Guide](/en/core-concepts/endpoints/#server-endpoints-api-routes).
\ No newline at end of file
+The function takes an [endpoint context](/en/reference/api-reference/#endpoint-context) and returns a [Response](https://developer.mozilla.org/en-US/docs/Web/API/Response). A powerful feature of SSR, API routes are able to securely execute code on the server. To learn more, see our [Endpoints Guide](/en/core-concepts/endpoints/#server-endpoints-api-routes).
+
+### Streaming
+
+Astro enables HTTP streaming for all pages that are not prerendered. This means that your page's content will be sent incrementally to the page as it resolves async functions and requests.
+
+You can `await` promises in the frontmatter of your page, or you can move those requests down into Astro components the page renders. The page will then send down the HTML it's capable of sending down without waiting for those async components, and then it will start rendering the components in the order they are found in your page's HTML one-by-one.
+
+For example, given a page that fetches some data:
+
+```astro title="src/pages/index.astro"
+---
+const data = await fetch('https://example.com')
+const otherData = await fetch('https//other-example.com')
+---
+<html>
+  <head>
+    <title>Astro example</title>
+  </head>
+  <body>
+    <p>{data.example}</p>
+    <p>{otherData.example}</p>
+  </body>
+</html>
```

Comment:
Generally, we tend to be a bit more concrete with our code examples, what do you think of making this one functional? 
```suggestion
---
const personResponse = await fetch('https://randomuser.me/api/');
const personData = await personResponse.json();
const randomPerson = personData.results[0];
const factResponse = await fetch('https://catfact.ninja/fact');
const factData = await factResponse.json();
---
<html>
  <head>
    <title>A name and a fact</title>
  </head>
  <body>
    <p>{randomPerson.name.first}</p>
    <p>{factData.fact}</p>
  </body>
</html>
```

Reply:
LOVE IT @Yan-Thomas